aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -342,376 +365,580 @@ void upsample_avx_bilinear_uint8(
   }
 }
 
-// https://gist.github.com/NicolasHug/47c97d731f05eaad5694c173849b86f5
 void ImagingResampleHorizontalConvolution8u4x(
     uint32_t* C10_RESTRICT lineOut0,
     uint32_t* C10_RESTRICT lineOut1,
     uint32_t* C10_RESTRICT lineOut2,
     uint32_t* C10_RESTRICT lineOut3,
+    int64_t out_xsize,
     const uint32_t* C10_RESTRICT lineIn0,
     const uint32_t* C10_RESTRICT lineIn1,
     const uint32_t* C10_RESTRICT lineIn2,
     const uint32_t* C10_RESTRICT lineIn3,
-    int xsize,
-    int* xbounds,
-    int16_t* kk,
+    const int64_t* idx_ptr_xmin,
+    const int64_t* idx_ptr_size,
+    const int16_t* kk,
     int kmax,
-    int coefs_precision) {
-  int xmin, xmax, x;
-  int16_t* k;
-
-  for (const auto xx : c10::irange(xsize)) {
-    xmin = xbounds[xx * 2 + 0];
-    xmax = xbounds[xx * 2 + 1];
-    k = &kk[xx * kmax];
-    x = 0;
-
-    __m256i sss0, sss1;
-    __m256i zero = _mm256_setzero_si256();
-    __m256i initial = _mm256_set1_epi32(1 << (coefs_precision - 1));
-    sss0 = initial;
-    sss1 = initial;
-
-    for (; x < xmax - 3; x += 4) {
-      __m256i pix, mmk0, mmk1, source;
-
-      mmk0 = _mm256_set1_epi32(*(int32_t*)&k[x]);
-      mmk1 = _mm256_set1_epi32(*(int32_t*)&k[x + 2]);
-
-      source = _mm256_inserti128_si256(
-          _mm256_castsi128_si256(_mm_loadu_si128((__m128i*)&lineIn0[x + xmin])),
-          _mm_loadu_si128((__m128i*)&lineIn1[x + xmin]),
-          1);
-      // clang-format off
-      pix = _mm256_shuffle_epi8(source, _mm256_set_epi8(
-        -1,7, -1,3, -1,6, -1,2, -1,5, -1,1, -1,4, -1,0,
-        -1,7, -1,3, -1,6, -1,2, -1,5, -1,1, -1,4, -1,0));
-      sss0 = _mm256_add_epi32(sss0, _mm256_madd_epi16(pix, mmk0));
-      pix = _mm256_shuffle_epi8(source, _mm256_set_epi8(
-        -1,15, -1,11, -1,14, -1,10, -1,13, -1,9, -1,12, -1,8,
-        -1,15, -1,11, -1,14, -1,10, -1,13, -1,9, -1,12, -1,8));
-      sss0 = _mm256_add_epi32(sss0, _mm256_madd_epi16(pix, mmk1));
-
-      source = _mm256_inserti128_si256(
-          _mm256_castsi128_si256(_mm_loadu_si128((__m128i*)&lineIn2[x + xmin])),
-          _mm_loadu_si128((__m128i*)&lineIn3[x + xmin]),
-          1);
-      pix = _mm256_shuffle_epi8(source, _mm256_set_epi8(
-        -1,7, -1,3, -1,6, -1,2, -1,5, -1,1, -1,4, -1,0,
-        -1,7, -1,3, -1,6, -1,2, -1,5, -1,1, -1,4, -1,0));
-      sss1 = _mm256_add_epi32(sss1, _mm256_madd_epi16(pix, mmk0));
-      pix = _mm256_shuffle_epi8(source, _mm256_set_epi8(
-        -1,15, -1,11, -1,14, -1,10, -1,13, -1,9, -1,12, -1,8,
-        -1,15, -1,11, -1,14, -1,10, -1,13, -1,9, -1,12, -1,8));
-      sss1 = _mm256_add_epi32(sss1, _mm256_madd_epi16(pix, mmk1));
+    unsigned int coefs_precision) {
+  // Interpolation horizontal pass processing together 4 vertical lines.
+  // - Input data format is RGBA with R,G,B,A being uint8, we can encode 4 values as a single uint32 value.
+  // - We split the size of weight vector for a given output index as a sum: K = n * 4 + m * 2 + k.
```

Comment:
The use of `k` or `K` here might be confusing with the `k` and `kk` naming in the code?